doc/build_man_files.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/sh
+
+#
+# general initialization cleanup
+#
+rm -f DoxygenLayout.xml
+rm -rf man_tmp
+mkdir -p man_tmp
+rm -rf man_html
+mkdir -p man_html
+
+#
+# Prefix the start of the 2 files
+#
+echo '    <tab type="usergroup" visible="yes" url="@ref manpage" title="Manual Pages" intro="">' > insert_file
+echo '/** @page manpage Manual Pages' > man_tmp/manpage.dox
+echo '  Here is a list of libcoap API manual pages, some of which have code examples:' >> man_tmp/manpage.dox
+echo '   <table class="directory">' >> man_tmp/manpage.dox
+
+FILES=`( cd ../man ; ls coap.txt.in ; ls coap_*.txt.in ; ls coap-*.txt.in )`
+ID=0
+ROW_EVEN=" class=\"even\""
+
+for FILE in $FILES ; do
+    BASE=`echo $FILE | cut -d. -f 1`
+    MANUAL=`cat ../man/$FILE | egrep -B 1 "^====" | head -1`
+    SUMMARY=`cat ../man/$FILE | egrep -B 2 "^SYNOPSIS" | sed 's/coap-//g' | cut -d\- -f2 | cut -c2- | head -1`
+
+    #
+    # Build the manual insert page
+    #
+    echo "/// @page man_$BASE $MANUAL" > man_tmp/$MANUAL.dox
+    echo "/// @htmlinclude $BASE.html $MANUAL" >> man_tmp/$MANUAL.dox
+
+    #
+    # Update insert_file
+    #
+    echo "      <tab type=\"user\" visible=\"yes\" url=\"@ref man_$BASE\" title=\"$MANUAL - $SUMMARY\" intro=\"\"/>" >> insert_file
+
+    #
+    # Update the summary man page
+    #
+    echo "   <tr id=\"row_${ID}_\"$ROW_EVEN>" >> man_tmp/manpage.dox
+    echo "   <td class=\"entry\" align=\"left\"> @ref man_$BASE </td><td class=\"desc\" align=\"left\">$MANUAL - $SUMMARY</td>" >> man_tmp/manpage.dox
+    echo "   </tr>" >> man_tmp/manpage.dox
+    let ID=$ID+1
```

Comment:
`let` is bash-specific while this script is invoked as `/bin/sh`:

```
make[2]: Entering directory '...libcoap/doc'
./build_man_files.sh: 46: ./build_man_files.sh: let: not found
./build_man_files.sh: 46: ./build_man_files.sh: let: not found
...
/usr/bin/doxygen Doxyfile
```

Reply:
I assume that ID=$(( $ID + 1 )) will do the trick - struggling to find a genuine bourne sh system..

Just doing a build with that update.